frame/bounties/src/lib.rs

```diff
@@ -0,0 +1,762 @@
+	Approved,
+	/// The bounty is funded and waiting for curator assignment.
+	Funded,
+	/// A curator has been proposed by the `ApproveOrigin`. Waiting for acceptance from the curator.
+	CuratorProposed {
+		/// The assigned curator of this bounty.
+		curator: AccountId,
+	},
+	/// The bounty is active and waiting to be awarded.
+	Active {
+		/// The curator of this bounty.
+		curator: AccountId,
+		/// An update from the curator is due by this block, else they are considered inactive.
+		update_due: BlockNumber,
+	},
+	/// The bounty is awarded and waiting to released after a delay.
+	PendingPayout {
+		/// The curator of this bounty.
+		curator: AccountId,
+		/// The beneficiary of the bounty.
+		beneficiary: AccountId,
+		/// When the bounty can be claimed.
+		unlock_at: BlockNumber,
+	},
+}
+
+decl_storage! {
+	trait Store for Module<T: Trait> as Treasury {
+
+		/// Number of bounty proposals that have been made.
+		pub BountyCount get(fn bounty_count): BountyIndex;
+
+		/// Bounties that have been made.
+		pub Bounties get(fn bounties):
+		map hasher(twox_64_concat) BountyIndex
+		=> Option<Bounty<T::AccountId, BalanceOf<T>, T::BlockNumber>>;
+
+		/// The description of each bounty.
+		pub BountyDescriptions get(fn bounty_descriptions): map hasher(twox_64_concat) BountyIndex => Option<Vec<u8>>;
+
+		/// Bounty indices that have been approved but not yet funded.
+		pub BountyApprovals get(fn bounty_approvals): Vec<BountyIndex>;
+	}
+}
+
+decl_event!(
+	pub enum Event<T>
+	where
+		Balance = BalanceOf<T>,
+		<T as frame_system::Trait>::AccountId,
+	{
+		/// New bounty proposal. [index]
+		BountyProposed(BountyIndex),
+		/// A bounty proposal was rejected; funds were slashed. [index, bond]
+		BountyRejected(BountyIndex, Balance),
+		/// A bounty proposal is funded and became active. [index]
+		BountyBecameActive(BountyIndex),
+		/// A bounty is awarded to a beneficiary. [index, beneficiary]
+		BountyAwarded(BountyIndex, AccountId),
+		/// A bounty is claimed by beneficiary. [index, payout, beneficiary]
+		BountyClaimed(BountyIndex, Balance, AccountId),
+		/// A bounty is cancelled. [index]
+		BountyCanceled(BountyIndex),
+		/// A bounty expiry is extended. [index]
+		BountyExtended(BountyIndex),
+	}
+);
+
+decl_error! {
+	/// Error for the treasury module.
+	pub enum Error for Module<T: Trait> {
+		/// Proposer's balance is too low.
+		InsufficientProposersBalance,
+		/// No proposal or bounty at that index.
+		InvalidIndex,
+		/// The reason given is just too big.
+		ReasonTooBig,
+		/// The bounty status is unexpected.
+		UnexpectedStatus,
+		/// Require bounty curator.
+		RequireCurator,
+		/// Invalid bounty value.
+		InvalidValue,
+		/// Invalid bounty fee.
+		InvalidFee,
+		/// A bounty payout is pending.
+		/// To cancel the bounty, you must unassign and slash the curator.
+		PendingPayout,
+		/// The tip cannot be claimed/closed because it's still in the countdown period.
```

Comment:
Check that all these errors are needed. This seems to imply it is an error for Tips

Reply:
Error enum "Premature" is valid and used in few extrinsic, rephrased the comment as below ...
"/// The bounties cannot be claimed/closed because it's still in the countdown period."